src/SprtTranslation.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Memoris
+ * Copyright (C) 2015  Jean LELIEVRE
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/**
+ * @file SprtTranslation.cpp
+ * @brief horizontal translation of a sprite
+ * @package utils
+ * @author Jean LELIEVRE <Jean.LELIEVRE@supinfo.com>
+ */
+
+#include "SprtTranslation.hpp"
+#include "Colors.hpp"
+
+using namespace utils;
+
+const uint8_t SprtTranslation::INTERVAL = 10;
+
+const float_t SprtTranslation::HRTL_INTERVAL = 10.f;
+
+/**
+ *
+ */
+SprtTranslation::SprtTranslation(
+    const float_t& hrtlPstnRef,
+    const float_t& vrtlPstnRef,
+    const float_t& hrtlSizeRef
+) : oriHrtlPstn(hrtlPstnRef)
+{
+
+    hrtlPstn = hrtlPstnRef;
+    vrtlPstn = vrtlPstnRef;
+    hrtlSize = hrtlSizeRef;
+
+    drct = 1;
+}
+
+/**
+ *
+ */
+void SprtTranslation::display(
+    utils::Context* ctx,
+    sf::Sprite& sprt,
+    const bool& animate
+)
+{
+    ctx->getWindow()->draw(sprt);
+
+    if (!animate)
```

Comment:
this code will be executed all the time, should be refactored inside a dedicated function, called only when the selected item changes